packages/arb-bridge-eth/contracts/arch/OneStepProof.sol

```diff
@@ -20,922 +20,772 @@ pragma solidity ^0.5.11;
-            return false;
-        }
-        uint256 a = val1.intVal;
-        uint256 b = val2.intVal;
-        uint256 c;
-        assembly {
-            c := sub(a, b)
+    function handleError(AssertionContext memory context) internal pure {
+        if (context.afterMachine.errHandlerHash == CODE_POINT_ERROR) {
+            context.afterMachine.setErrorStop();
+        } else {
+            context.afterMachine.instructionStackHash = context
+                .afterMachine
+                .errHandlerHash;
         }
-        machine.addDataStackInt(c);
-        return true;
     }
 
-    function executeDivInsn(
-        Machine.Data memory machine,
-        Value.Data memory val1,
-        Value.Data memory val2
-    ) internal pure returns (bool) {
-        if (!val1.isInt() || !val2.isInt()) {
-            return false;
-        }
-        uint256 a = val1.intVal;
-        uint256 b = val2.intVal;
-        if (b == 0) {
-            return false;
-        }
-        uint256 c;
-        assembly {
-            c := div(a, b)
-        }
-        machine.addDataStackInt(c);
-        return true;
+    function handleOpcodeError(AssertionContext memory context) internal pure {
+        context.handleError();
+        // Also clear the stack and auxstack
+        context.stack.length = 0;
+        context.auxstack.length = 0;
     }
 
-    function executeSdivInsn(
-        Machine.Data memory machine,
-        Value.Data memory val1,
-        Value.Data memory val2
-    ) internal pure returns (bool) {
-        if (!val1.isInt() || !val2.isInt()) {
-            return false;
-        }
-        uint256 a = val1.intVal;
-        uint256 b = val2.intVal;
-        if (b == 0) {
-            return false;
-        }
-        uint256 c;
-        assembly {
-            c := sdiv(a, b)
-        }
-        machine.addDataStackInt(c);
-        return true;
-    }
+    function initializeExecutionContext(
+        Value.Data memory inbox,
+        bytes32 messagesAcc,
+        bytes32 logsAcc,
+        bytes memory proof
+    ) internal pure returns (AssertionContext memory) {
+        uint8 stackCount = uint8(proof[0]);
+        uint8 auxstackCount = uint8(proof[1]);
+
+        // Leave some extra space for values pushed on the stack in the proofs
+        Value.Data[] memory stackVals = new Value.Data[](stackCount + 4);
+        Value.Data[] memory auxstackVals = new Value.Data[](auxstackCount + 4);
+        uint256 offset = 2;
+        for (uint256 i = 0; i < stackCount; i++) {
+            (offset, stackVals[i]) = Marshaling.deserialize(proof, offset);
+        }
+        for (uint256 i = 0; i < auxstackCount; i++) {
+            (offset, auxstackVals[i]) = Marshaling.deserialize(proof, offset);
+        }
+        Machine.Data memory mach;
+        (offset, mach) = Machine.deserializeMachine(proof, offset);
+
+        uint8 immediate = uint8(proof[offset]);
+        uint8 opCode = uint8(proof[offset + 1]);
```

Comment:
Adding 2 to offset here is safer, to reduce the odds of error if the code below changes.

Reply:
Added